TutByParametrizedTests/Task2tests/Task2tests/Helpers/WaitHelper.cs

```diff
@@ -0,0 +1,101 @@
+﻿using OpenQA.Selenium;
+using OpenQA.Selenium.Support.UI;
+using System;
+
+namespace Training.Helpers
+{
+    public static class WaitHelper
+    {
+        private static readonly IWebDriver Driver = DriverGenerator.GetInstance();
+
+        public static void WaitElementExist(By by)
```

Comment:
Looks like all your methods have to bool return value, but not void in this class. Selenium waiters need to return current result of condition verifying. It means that if you wait until some action should be completed then you need to return the result of your waiter to get a chance to compare it in Assert method. For example, you have WaitElementExist() method. It provides information if element, which searching based on By parameter, exists or not. So, in test method it needs to be written something like this: `Assert.IsTrue(WaitHelper.WaitElementExist(userName), "The username element doesn't exist");`